### DISCUSSION TITLE
Does hls4ml work on Windows?

### DISCUSSION THREAD

**Discussion by janarbek** [fastmachinelearning/hls4ml#899] [General]
Hi All,

I have seen that it is not supported on Windows, but wanted to verify if hls4ml works on Windows? Anyone tried? What it takes to make it run on Windows?

**Comment by vloncar** [comment]
Compilation of projects for simulation (`model.compile()`, `model.predict()` etc) fails due to compiler not being available and setting up an environment for it via Cygwin/MingW is cumbersome to support. Furthermore the generated HLS project doesn't compile with Vivado HLS tools due to issues in their toolchain. We didn't try the more recent Vitis HLS. It is unlikely we will support Windows in any official way since none of the developers use it. You can use WSL to setup Linux within windows where hls4ml works as expected, but you may still run into issues with synthesis if you use Windows installation of Vivado/Vitis

**Comment by janarbek** [comment]
Hii Vladimir,

I have seen a recent tutorial from here:
https://nssmic.ieee.org/2023/program/

I am not sure if this tutorial is given to your group, but I was under the
impression that we can use Windows (in the prerequisite of the tutorial it
is listed that Windows or linux is needed?

If we can use Windows, that would be great -- any help or resources will be
appreciated.

Thanks,
J.

**Comment by jmitrevs** [comment]
The tutorial is done on the cloud so the computer doesn’t matter. Jovan On Nov 1, 2023, at 19:57, janarbek ***@***.***> wrote:﻿
Hii Vladimir,



I have seen a recent tutorial from here:

https://nssmic.ieee.org/2023/program/



I am not sure if this tutorial is given to your group, but I was under the

impression that we can use Windows (in the prerequisite of the tutorial it

is listed that Windows or linux is needed?



If we can use Windows, that would be great -- any help or resources will be

appreciated.



Thanks,

J.

**Comment by janarbek** [comment]
Thank you for the prompt response, now I understand.

I am wondering if we can access the tutorial on the cloud? I tried to use
the link (https://tutorials.fastmachinelearning.org/), but it does not work.

Thanks,
J.

**Comment by vloncar** [comment]
I believe that link is working during the tutorial, we don't spin up cloud resources to be idle outside of the tutorials. Just to clarify, only one step of the flow from the model to hardware doesn't work on windows, the C++ compilation that is part of simulation and C synthesis. Once you have the bitfile produced you can move to your platform of choice, which is what I think is done in hands-on with the Pynq board. 

Perhaps the organizers, @elhamekhoda and @ben-hawks can elaborate?

**Comment by ben-hawks** [comment]
Hi @janarbek! @vloncar is correct, all the compilation itself is done in the cloud instance for the tutorial, then the final files are moved over to the board via Jupyerhub, scp, etc. to further run locally on the Pynq-Z2. 

The course requirements (windows or linux) are primarily so that if something happens, it's possible to debug/program the hardware with what the student has. 

If you're interested in trying to run/access the tutorial material outside of the course, it's available in a docker container (with or without Vivado) you can deploy where convenient: https://github.com/fastmachinelearning/hls4ml-tutorial/tree/main#docker-with-vivado

**Comment by janarbek** [comment]
Hi Ben,

So I used WSL 2 and docker to run the docker image. I was able to set up
everything successfully -- and now I am in the last stage where I have some
questions.

When I run the following command, I am getting the screen shown below. I
opened the jupyter notebook, but I do not see the examples (part 1- part 4)
as suggested by tutorial slides.

docker run -p 8888:8888
ghcr.io/fastmachinelearning/hls4ml-tutorial/hls4ml-0.7.1-vivado-2019.2:latest

This is what I see after running the docker.

[image: image.png]

This is what jupyter notebook looks like. So my question is where are the
example files (part 1 to part4)?

[image: image.png]
